network/transport/v2/protocol.go

```diff
@@ -71,22 +116,135 @@ func (p protocol) UnwrapMessage(envelope interface{}) interface{} {
 func (p protocol) Configure(_ transport.PeerID) {
 }
 
-func (p protocol) Start() {
+func (p *protocol) setupNatsHandler() error {
+	conn, err := p.natsConnectHandler(
+		p.config.Nats.Hostname,
+		p.config.Nats.Port,
+		time.Duration(p.config.Nats.Timeout)*time.Second,
+	)
+	if err != nil {
+		return err
+	}
+
+	js, err := conn.JetStream()
+	if err != nil {
+		return err
+	}
+
+	if _, err = js.Subscribe(events.PrivateTransactionsSubject, func(msg *nats.Msg) {
+		if err := p.handlePrivateTx(msg); err != nil {
+			logrus.Errorf("failed to handle private transaction: %v", err)
```

Comment:
make sure we log something we can recognize the TX from (i'd say `ref`)

Reply:
as trace logging

Reply:
We can't do that here as we haven't parsed the NATS message yet. I added `ref` in the custom errors in `handlePrivateTx(...)` instead

Reply:
Not just TRACE logging, we'll have to make sure it gets logged with default production settings imo. Enabling DEBUG/TRACE on production systems was often not allowed in previous environments, because it often contained sensitive information. And it yielded so much, that I hated having to spit through it trying to find out what went wrong.

I generally find Go errors OK (or maybe just in our Nuts node) for finding out _what_ went wrong (e.g. parsing error), but generally poor for finding out _where_ (record XYZ). Especially important when there's a lot of options (e.g. thousands of TXs)